### PR TITLE
issue 5- Address entity and Customer entity added

### PR DESCRIPTION
**Pull request by Abhikeshgore**
com/customerservice/entities/Customer.java- Added External_ID For Customers, Made changes in Join 
` @OneToMany(targetEntity = **Address.class** ,cascade =  CascadeType.ALL)
    @JoinColumn(name = "**ca_fk**",referencedColumnName = "id")
    @JsonIgnoreProperties("customer")
    private List<Address> address;`.
ca_fk= Customer_Address foreign key.
com/customerservice/entities/Address.java- Added AddressExternal_ID For Address.
 com/customerservice/services/impl/AddressServiceImpl.java - Implementation of method to Update Address
com/customerservice/services/impl/CustomerServiceImpl.java -Implementation of  method to Add New Customer Record And also Method to find CustomerByExternal_ID

### CODE REVIEW

File: src/main/java/com/customerservice/services/impl/AddressServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        this.customerConverter=customerConverter;
+        this.addressConverter=addressConverter;
+
+    }
+
+    @Override
+    public CustomerDto updateAddress(UUID externalId, AddressDto addressDto) {
+        List<Address> addressList = new ArrayList<>();
+        addressList.add(addressConverter.DtoToEntity(addressDto));
+        Customer customer = customerRepo.findById(externalId).orElseThrow(() -> new CustomerNotFoundException(ErrorConstant.CUSTOMER_NOT_FOUND));
```
Comment (pranay-gore) [inline src/main/java/com/customerservice/services/impl/AddressServiceImpl.java:40]: You can check this on the first line itself. You don't have to convert address entity if customer itself is not there. So you can save some time and improve performance. Once customer is there, then you can convert address DTO to entity

**Comment by pranay-gore** [comment]
CustomerServiceApplicationTests seems to be failing. I tried command man clean install and it failed

**Review by pranay-gore — APPROVED** [review]
Please update the code according to the small comments given. Once you update them, then you can you merge it to main branch

File: pom.xml [hunk 2]
```diff
@@ -26,6 +26,21 @@
...
 
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-validation</artifactId>
+		</dependency>
+		<!-- https://mvnrepository.com/artifact/io.swagger/swagger-annotations -->
+		<dependency>
+			<groupId>io.swagger</groupId>
+			<artifactId>swagger-annotations</artifactId>
+			<version>1.5.22</version>
```
[the remark this answers is not in the document]
Comment (Abhikeshgore) [reply pom.xml:37]: `<properties>
		<java.version>17</java.version>
		<swagger-ui.version>1.5.22</swagger-ui.version>
		<h2database.version>2.2.220</h2database.version>
	</properties>`

**all versions mentioned at one place under property variable**

File: src/main/java/com/customerservice/controllers/AddressController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+import com.customerservice.dtos.CustomerDto;
+import com.customerservice.services.AddressService;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/customers/{id}/addresss")
```
[the remark this answers is not in the document]
Comment (Abhikeshgore) [reply src/main/java/com/customerservice/controllers/AddressController.java:16]: Changed addresss to addresses

File: src/main/java/com/customerservice/services/impl/AddressServiceImpl.java (hunk 1 above)
Comment (Abhikeshgore) [reply src/main/java/com/customerservice/services/impl/AddressServiceImpl.java:40]: yes

Comment (Abhikeshgore) [reply src/main/java/com/customerservice/services/impl/AddressServiceImpl.java:40]: made changes here...on first line itself , I am checking for if The customer is there or not there

**Pull request merged**